core_backend/app/question_answer/routers.py

```diff
@@ -212,3 +215,53 @@ async def feedback(
                 )
             },
         )
+
+
+@router.post("/content-feedback")
+async def content_feedback(
+    feedback: ContentFeedback, asession: AsyncSession = Depends(get_async_session)
+) -> JSONResponse:
+    """
+    Feedback endpoint used to capture user feedback on specific content
+    """
+
+    is_matched = await check_secret_key_match(
+        feedback.feedback_secret_key, feedback.query_id, asession
+    )
+
+    if is_matched is False:
+        return JSONResponse(
+            status_code=400,
+            content={
+                "message": f"Secret key does not match query id : {feedback.query_id}"
+            },
+        )
+    else:
+        try:
+            feedback_db = await save_content_feedback_to_db(feedback, asession)
+        except IntegrityError as e:
```

Comment:
If you give it the wrong content_id, then it will result in a Foreign Key violation. This captures that.

Reply:
Ah I was thinking that we are checking that this content was indeed returned in the response for this query. But we are just checking that it's a valid content ID it seems?